File: 2399-check-distances-between-same-letters/2399-check-distances-between-same-letters.py
class Solution:
    def checkDistances(self, s: str, distance: List[int]) -> bool:
        dicts = {}
        for i in range(len(s)):
            if s[i] in dicts:
                dicts[s[i]].append(i)
            else:
                dicts[s[i]] = [i]
        print(dicts)
        for i in dicts:
            x = ord(i)-ord('a')
            arr = dicts[i]
            diff = arr[1] - arr[0]
            if distance[x] != diff - 1:
                return False
        return True
            

File: 2399-check-distances-between-same-letters/2399-check-distances-between-same-letters.java
class Solution {
    public boolean checkDistances(String s, int[] distance) {
        Map<Character, List<Integer>> map = new HashMap<>();
        for(int i=0;i<s.length();i++){
            if (map.containsKey(s.charAt(i))){ 
                List<Integer> l = map.get(s.charAt(i));
                l.add(i);
                map.put(s.charAt(i), l);
            } else {
                List<Integer> l = new ArrayList<Integer>();
                l.add(i);
                map.put(s.charAt(i), l);
            }
        }
        for (Map.Entry<Character, List<Integer>> mE : map.entrySet()) {
            Character c = mE.getKey();
            List<Integer> index = mE.getValue();
            int x = c - 'a';
            System.out.println(x);
            int diff = index.get(1) - index.get(0);
            if (diff-1 != distance[x]){
                return false;
            }
        }
        return true;
    }
}